docs/guide/rpc-api.md

```diff
@@ -1,5 +1,309 @@
+This method is specified by [EIP-747](https://eips.ethereum.org/EIPS/eip-747).
+:::
+
+```typescript
+interface WatchAssetParams {
+  type: 'ERC20', // In the future, other standards will be supported
+  options: {
+    address: string, // The address of the token contract
+    'symbol': string, // A ticker symbol or shorthand, up to 5 characters
+    decimals: number, // The number of token decimals
+    image: string, // A string url of the token logo
+  },
+}
+
+ethereum.request({
+  method: 'wallet_watchAsset',
+  params: [ WatchAssetParams ]
+}): Promise<boolean>
+```
+
+Requests that the user tracks the token in MetaMask.
+Returns a `boolean` indicating if the token was successfully added.
+
+Most Ethereum wallets support some set of tokens, usually from a centrally curated registry of tokens.
+`wallet_watchAsset` enables web3 application developers to ask their users to track tokens in their wallets, at runtime.
+Once added, the token is indistinguishable from those added via legacy methods, such as a centralized registry.
+
+### eth_getEncryptionPublicKey
+
+```typescript
+// both types are simply hexadecimal strings
+type EthereumAddress: string;
+type EncryptionPublicKey: string;
+
+ethereum.request({
+  method: 'eth_getEncryptionPublicKey',
+  params: [ EthereumAddress ]
+}): Promise<EncryptionPublicKey>
+```
+
+Requests that the user shares their encryption public key.
+Returns a Promise that resolve to the encryption public key, or rejects if the user denied the request.
+
+The public key is computed from entropy associated with the specified user account, using the [`nacl`](https://github.com/dchest/tweetnacl-js) implementation of the `X25519_XSalsa20_Poly1305` algorithm.
+
+#### Example
+
+```javascript
+let encryptionPublicKey;
+
+ethereum
+  .request({
+    method: 'eth_getEncryptionPublicKey',
+    params: [accounts[0]], // you must have access to the specified account
+  })
+  .then((result) => {
+    encryptionPublicKey = result;
+  })
+  .catch((error) => {
+    if (error.code === 4001) {
+      // EIP-1193 userRejectedRequest error
+      console.log('We can encrypt anything without the key.');
+    } else {
+      console.error(error);
+    }
+  });
+```
+
+#### Encrypting
+
+The point of the encryption key is of course to encrypt things.
+Here's an example of how to encrypt a message using [`eth-sig-util`](https://github.com/MetaMask/eth-sig-util):
+
+```javascript
+const ethUtil = require('ethereumjs-util');
+
+const encryptedMessage = ethUtil.bufferToHex(
+  Buffer.from(
+    JSON.stringify(
+      sigUtil.encrypt(
+        encryptionPublicKey,
+        { data: 'Hello world!' },
+        'x25519-xsalsa20-poly1305'
+      )
+    ),
+    'utf8'
+  )
+);
+```
+
+### eth_decrypt
+
+```typescript
+// these two types are simply hexadecimal strings
+type EthereumAddress: string;
+type EncryptedMessage: string;
+// the decrypted message is the originally encrypted string
+type DecryptedMessage: string;
+
+ethereum.request({
```

Comment:
I get that the goal of this is to inform the user of the types, but this wouldn't compile because its a mixture of a function invocation with a type signature for that function invocation. You'd need to make ethereum.request a parameterized function eg:

```ts
function request<ParamsType, ReturnType>({ method: string, params: ParamsType }): Promise<ReturnType> {
  // implementation
}

request<[EncryptedMessage, EthereumAddress], DecryptedMessage>({ method: 'eth_decrypt', params: ['0xasdf1a7=', '0xa00a..324r3']
}).then(decryptedMessage => { // do something with decryption });
```

If we have an explainer somewhere that the examples are just for showing types and not real code, then we can just leave it as is maybe?

Reply:
Hmm, yeah, the goal is mostly to be informative, and I think for most devs, the parameterized function obscures more than it explains, even though it compiles 🤔 

I'm inclined to leave a notice, depending on what we all think. It is a bit ugly to have it listed as TS that's just... syntactically invalid.

Reply:
alternatively...

```ts
// Create an intermediary typesafe function
function ethDecrypt(params: [EncryptedMessage, EthereumAddress]): Promise<DecryptedMessage> {
  return ethereum.request({ method: 'eth_decrypt', params })
}
```

might be cleaner in regards to not relying on Algebraic data types, compiles, etc.

Reply:
Hm, or, alternatively alternatively, I suppose we do what most JSON-RPC APIs I’ve seen do, e.g. https://kodi.wiki/view/JSON-RPC_API/v10

It’s the same for the Ethereum JSON-RPC API listed on eth.wiki.

Reply:
@Gudahtt @brad-decker this is the last thing we need to check off for this PR. See: 3dc33db

The addition of `.prettierrc.js` was necessary to create an exception for a particular snippet that used a reserved TypeScript keyword (`symbol`) in an interface.